packages/mobile/src/components/CurrencyDisplay.tsx

```diff
@@ -2,61 +2,128 @@ import colors from '@celo/react-components/styles/colors'
 import fontStyles from '@celo/react-components/styles/fonts'
 import BigNumber from 'bignumber.js'
 import * as React from 'react'
-import { StyleSheet, Text, View } from 'react-native'
+import { StyleProp, StyleSheet, Text, TextStyle, View } from 'react-native'
+import { MoneyAmount } from 'src/apollo/types'
 import { CURRENCIES, CURRENCY_ENUM } from 'src/geth/consts'
-import { LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { LocalCurrencyCode, LocalCurrencySymbol } from 'src/localCurrency/consts'
+import { useDollarsToLocalAmount, useLocalCurrencyCode } from 'src/localCurrency/hooks'
 import { getMoneyDisplayValue } from 'src/utils/formatting'
 
+export enum DisplayType {
+  Default,
+  Big, // symbol displayed as superscript
+}
+
 interface Props {
-  amount: BigNumber
-  size: number
-  type: CURRENCY_ENUM
-  currencySymbol?: LocalCurrencySymbol | null
+  type: DisplayType
+  amount: MoneyAmount
+  size: number // only used for DisplayType.Big
+  useColors: boolean
+  hideSymbol: boolean
+  formatAmount: (amount: BigNumber.Value) => string
```

Comment:
Can the formatAmount be determined from the DisplayType? Or maybe made optional? Just wondering if we can keep as much currency formatting related logic encapsulated in this file as possible.

Reply:
Though it's not visible here, it's optional (see `defaultProps`) and used to format some network fee amounts. 
DisplayType is currently used for determining how we display the symbol (superscript vs normal, maybe I should rename that too).
I thought about providing another enum for the format, but decided to leave it as is with a default implementation until I refactor more of the code related to local currencies.
I agree it should encapsulate all formatting logic too.